Tasks/Maven/pmdformaven.ts

```diff
@@ -10,25 +10,27 @@ import trm = require('vsts-task-lib/toolrunner');
 
 import ar = require('./analysisresult');
 
+export const toolName:string = 'PMD';
+
 // Adds PMD goals, if selected by the user
 export function applyPmdArgs(mvnRun: trm.ToolRunner):void {
-    mvnRun.arg(['jxr:jxr', 'pmd:pmd']); // JXR source code cross-referencing is a required pre-goal for PMD
+    mvnRun.arg(['pmd:pmd', '-DlinkXRef=false']); // Turn off cross-referencing to reduce Maven error output
 }
 
 // Extract analysis results from PMD output file.
-// Takes the current working directory and returns an AnalysisResult data class.
+// Takes the working directory (should contain pom.xml and target/) and returns an AnalysisResult data class.
 // Task fails if the HTML or XML outputs were not found.
 // @param sourcesDirectory - The absolute location of the root source directory.
-export function processPmdOutput(sourcesDirectory:string) : ar.AnalysisResult {
+export function processPmdOutput(rootDir:string) : ar.AnalysisResult {
```

Comment:
No longer assumes we are working directly in the agent sources directory, but we still expect a /target/ dir.